compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -1347,7 +1347,7 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             return;
         };
         // Try to find predicates on *generic params* that would allow copying `ty`
-        let ocx = ObligationCtxt::new(self.infcx);
+        let ocx = ObligationCtxt::new_with_diagnostics(self.infcx);
```

Comment:
we use `new_with_diagnostics` because we care about leaves instead of the root goal?

Reply:
Yes, I believe so.